Tests/LibJS/test-js.cpp

```diff
@@ -58,6 +58,42 @@ TESTJS_GLOBAL_FUNCTION(get_weak_map_size, getWeakMapSize)
     return JS::Value(weak_map->values().size());
 }
 
+TESTJS_GLOBAL_FUNCTION(force_gc, forceGC)
+{
+    if (!vm.argument(0).is_string()) {
+        vm.throw_exception<JS::TypeError>(global_object, JS::ErrorType::NotA, "String");
+        return {};
+    }
+
+    auto& variable_name = vm.argument(0).as_string();
+    if (vm.exception())
+        return {};
+
+    // In native functions we don't have a lexical environment so get the outer via the execution stack.
+    auto outer_environment = vm.execution_context_stack().last_matching([&](auto& execution_context) {
+        return execution_context->lexical_environment != nullptr;
+    });
+    if (!outer_environment.has_value())
+        return {};
+
+    auto variable = outer_environment.value()->lexical_environment->get_from_environment(variable_name.string());
+    if (!variable.has_value()) {
+        global_object.vm().throw_exception<JS::ReferenceError>(global_object, JS::ErrorType::UnknownIdentifier, variable_name.string());
+        return {};
+    }
+
+    if (variable->value.is_object()) {
+        vm.heap().add_inverse_root(&variable->value.as_object());
+        vm.heap().collect_garbage();
+    } else {
+        global_object.vm().throw_exception<JS::TypeError>(global_object, JS::ErrorType::NotAnObject, String::formatted("Variable with name {}", variable_name.string()));
+        return {};
+    }
+
+    outer_environment.value()->lexical_environment->put_into_environment(variable_name.string(), { JS::js_undefined(), variable->declaration_kind });
```

Comment:
I wonder if `delete_from_environment` would be more appropriate here? Since currently were just replacing it with undefined, so accessing it will still be fine, but if we delete it we essentially remove it from the scope, meaning accesses to it will result in an unknown identifier exception, not sure which is preferred

Reply:
Normally you're not allowed to delete a variable like that I think but maybe that makes it very clear it has been cleaned up.